# Disable Glic ("Gemini Live in Chrome"), ScreenAI, and associated AI-based
# features.

diff -up chromium-156.0.8078.35/chrome/browser/accessibility/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/accessibility/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/accessibility/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/accessibility/BUILD.gn	2026-10-20 17:09:47.117654483 +0200
@@ -59,7 +59,6 @@ source_set("accessibility") {
     public_deps += [
       "//chrome/browser/accessibility/live_caption",
       "//chrome/browser/profiles:profile_manager",
-      "//chrome/browser/screen_ai:screen_ai_install_state",
       "//components/live_caption",
       "//components/pref_registry",
       "//ui/accessibility/platform",
@@ -88,8 +87,6 @@ source_set("impl") {
 
   if (!is_android) {
     sources += [
-      "ax_main_node_annotator_controller.cc",
-      "ax_main_node_annotator_controller_factory.cc",
       "embedded_a11y_extension_loader.cc",
       "invert_bubble_prefs.cc",
       "page_colors_controller.cc",
@@ -99,7 +96,6 @@ source_set("impl") {
       "//chrome/browser/accessibility/live_caption:impl",
       "//chrome/browser/extensions",
       "//chrome/browser/on_device_translation",
-      "//chrome/browser/screen_ai:screen_ai_service_router_factory",
       "//chrome/browser/ui",
       "//chrome/browser/ui/browser_window",
       "//chrome/browser/ui/views/frame:browser_view_header",
@@ -246,9 +242,7 @@ if (!is_android) {
     }
 
     deps += [
-      "//chrome/browser/screen_ai:screen_ai_service_router_factory",
       "//chrome/browser/ui/views/frame:browser_view_header",
-      "//services/screen_ai/public/cpp:utilities",
     ]
 
     if (enable_screen_ai_browsertests) {
diff -up chromium-156.0.8078.35/chrome/browser/accessibility/live_caption/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/accessibility/live_caption/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/accessibility/live_caption/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/accessibility/live_caption/BUILD.gn	2026-10-20 17:09:47.118038780 +0200
@@ -44,7 +44,6 @@ source_set("impl") {
     "//chrome/browser/accessibility",
     "//chrome/browser/extensions",
     "//chrome/browser/on_device_translation",
-    "//chrome/browser/screen_ai:screen_ai_service_router_factory",
     "//chrome/browser/ui",
     "//chrome/browser/ui/browser_window",
     "//components/live_caption",
diff -up chromium-156.0.8078.35/chrome/browser/accessibility/tree_fixing/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/accessibility/tree_fixing/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/accessibility/tree_fixing/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/accessibility/tree_fixing/BUILD.gn	2026-10-20 17:09:47.118319555 +0200
@@ -63,11 +63,9 @@ source_set("internal") {
   deps = [
     "//base",
     "//chrome/browser/profiles:profile",
-    "//chrome/browser/screen_ai:screen_ai_service_router_factory",
     "//components/paint_preview/browser",
     "//components/paint_preview/common",
     "//components/paint_preview/public",
     "//content/public/browser",
-    "//services/screen_ai/public/mojom",
   ]
 }
diff -up chromium-156.0.8078.35/chrome/browser/actor/actor_keyed_service.cc.disable-ai chromium-156.0.8078.35/chrome/browser/actor/actor_keyed_service.cc
--- chromium-156.0.8078.35/chrome/browser/actor/actor_keyed_service.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/actor/actor_keyed_service.cc	2026-10-20 17:09:47.118847319 +0200
@@ -870,13 +870,6 @@ bool ActorKeyedService::IsActiveOnTab(co
 
 ActorTask* ActorKeyedService::GetTaskFromTab(
     const tabs::TabInterface& tab) const {
-  tabs::TabHandle handle = tab.GetHandle();
-  for (const auto& [task_id, task] : active_tasks_) {
-    if (task->HasTab(handle)) {
-      return task.get();
-    }
-  }
-
   return nullptr;
 }
 
diff -up chromium-156.0.8078.35/chrome/browser/browser_process.h.disable-ai chromium-156.0.8078.35/chrome/browser/browser_process.h
--- chromium-156.0.8078.35/chrome/browser/browser_process.h.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/browser_process.h	2026-10-20 17:09:47.119445405 +0200
@@ -331,8 +331,6 @@ class BrowserProcess {
   virtual void set_usb_system_tray_icon_for_test(
       std::unique_ptr<UsbSystemTrayIcon> icon) = 0;
 
-  virtual speech::SpeechRecognitionSmallExpertModelInstaller*
-  speech_recognition_small_expert_model_installer() = 0;
 #endif
 
   // Obtain the browser instance of OSCryptAsync, which should be used for data
diff -up chromium-156.0.8078.35/chrome/browser/browser_process_impl.cc.disable-ai chromium-156.0.8078.35/chrome/browser/browser_process_impl.cc
--- chromium-156.0.8078.35/chrome/browser/browser_process_impl.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/browser_process_impl.cc	2026-10-20 17:09:47.120093661 +0200
@@ -85,7 +85,6 @@
 #include "chrome/browser/serial/serial_policy_allowed_ports.h"
 #include "chrome/browser/shell_integration.h"
 #include "chrome/browser/site_isolation/prefs_observer.h"
-#include "chrome/browser/speech/speech_recognition_small_expert_model_installer.h"
 #include "chrome/browser/ssl/secure_origin_prefs_observer.h"
 #include "chrome/browser/startup_data.h"
 #include "chrome/browser/status_icons/status_tray.h"
@@ -261,12 +260,10 @@ void OnLocalStatePrefsLoaded();
 #if BUILDFLAG(IS_CHROMEOS)
 #include "chrome/browser/chromeos/extensions/telemetry/chromeos_telemetry_extensions_browser_api_provider.h"
 #include "chrome/browser/hid/hid_pinned_notification.h"
-#include "chrome/browser/screen_ai/screen_ai_downloader_chromeos.h"
 #include "chrome/browser/usb/usb_pinned_notification.h"
 #include "components/crash/core/app/crashpad.h"
 #elif !BUILDFLAG(IS_ANDROID)
 #include "chrome/browser/hid/hid_status_icon.h"
-#include "chrome/browser/screen_ai/screen_ai_downloader_non_chromeos.h"
 #include "chrome/browser/usb/usb_status_icon.h"
 #include "components/enterprise/browser/controller/chrome_browser_cloud_management_controller.h"
 #endif
@@ -1368,12 +1365,6 @@ GlobalFeatures* BrowserProcessImpl::GetF
   return features_.get();
 }
 
-#if !BUILDFLAG(IS_ANDROID)
-speech::SpeechRecognitionSmallExpertModelInstaller*
-BrowserProcessImpl::speech_recognition_small_expert_model_installer() {
-  return speech_recognition_small_expert_model_installer_.get();
-}
-#endif
 
 DownloadRequestLimiter* BrowserProcessImpl::download_request_limiter() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
@@ -1715,11 +1706,6 @@ void BrowserProcessImpl::PreMainMessageL
   soda_installer_impl_ = std::make_unique<speech::SodaInstallerImplChromeOS>();
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
-#if !BUILDFLAG(IS_ANDROID)
-  screen_ai_download_ = screen_ai::ScreenAIInstallState::Create();
-  speech_recognition_small_expert_model_installer_ =
-      std::make_unique<speech::SpeechRecognitionSmallExpertModelInstaller>();
-#endif
 
   base::FilePath user_data_dir;
   bool result = base::PathService::Get(chrome::DIR_USER_DATA, &user_data_dir);
diff -up chromium-156.0.8078.35/chrome/browser/browser_process_impl.h.disable-ai chromium-156.0.8078.35/chrome/browser/browser_process_impl.h
--- chromium-156.0.8078.35/chrome/browser/browser_process_impl.h.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/browser_process_impl.h	2026-10-20 17:09:47.120584088 +0200
@@ -58,9 +58,6 @@ class SiteIsolationPrefsObserver;
 class SystemNotificationHelper;
 class StartupData;
 
-namespace speech {
-class SpeechRecognitionSmallExpertModelInstaller;
-}  // namespace speech
 
 namespace breadcrumbs {
 class ApplicationBreadcrumbsLogger;
@@ -101,9 +98,6 @@ namespace speech {
 class SodaInstaller;
 }  // namespace speech
 
-namespace screen_ai {
-class ScreenAIInstallState;
-}  // namespace screen_ai
 
 namespace supervised_user {
 class DeviceParentalControls;
@@ -241,8 +235,6 @@ class BrowserProcessImpl : public Browse
   UsbSystemTrayIcon* usb_system_tray_icon() override;
   void set_usb_system_tray_icon_for_test(
       std::unique_ptr<UsbSystemTrayIcon> icon) override;
-  speech::SpeechRecognitionSmallExpertModelInstaller*
-  speech_recognition_small_expert_model_installer() override;
 #endif
 
   os_crypt_async::OSCryptAsync* os_crypt_async() override;
@@ -461,12 +453,6 @@ class BrowserProcessImpl : public Browse
   // to ensure that SodaInstallerImpl gets destructed first.
   std::unique_ptr<speech::SodaInstaller> soda_installer_impl_;
 
-  // Used to download Screen AI on demand and keep track of the library
-  // availability.
-  std::unique_ptr<screen_ai::ScreenAIInstallState> screen_ai_download_;
-
-  std::unique_ptr<speech::SpeechRecognitionSmallExpertModelInstaller>
-      speech_recognition_small_expert_model_installer_;
 #endif
 
   std::unique_ptr<BrowserProcessPlatformPart> platform_part_;
diff -up chromium-156.0.8078.35/chrome/browser/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/BUILD.gn.disable-ai	2026-10-20 17:09:46.910549833 +0200
+++ chromium-156.0.8078.35/chrome/browser/BUILD.gn	2026-10-20 17:09:47.115553053 +0200
@@ -2605,7 +2605,6 @@ source_set("browser") {
       "//chrome/browser/profile_resetter:impl",
       "//chrome/browser/record_replay",
       "//chrome/browser/record_replay:impl",
-      "//chrome/browser/screen_ai:screen_ai_service_router_factory",
       "//chrome/browser/sharing_hub",
       "//chrome/browser/sharing_hub:impl",
       "//chrome/browser/smart_card",
@@ -2654,7 +2653,6 @@ source_set("browser") {
       "//chrome/browser/ui/waap:window_metrics_manager",
       "//chrome/browser/ui/web_applications:impl",
       "//chrome/browser/ui/webui/app_service_internals",
-      "//chrome/browser/ui/webui/autofill_ml_internals",
       "//chrome/browser/ui/webui/color_pipeline_internals",
       "//chrome/browser/ui/webui/commerce",
       "//chrome/browser/ui/webui/commerce:impl",
diff -up chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders.cc.disable-ai chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders.cc
--- chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders.cc	2026-10-20 17:09:47.120968490 +0200
@@ -91,8 +91,6 @@
 
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
     BUILDFLAG(IS_CHROMEOS)
-#include "chrome/browser/screen_ai/screen_ai_service_router.h"
-#include "chrome/browser/screen_ai/screen_ai_service_router_factory.h"
 #include "chrome/browser/web_applications/sub_apps/sub_apps_service_impl.h"
 #endif
 
@@ -387,27 +385,6 @@ void BindMediaFoundationPreferences(
 }
 #endif  // BUILDFLAG(IS_WIN)
 
-#if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_WIN)
-void BindScreenAIAnnotator(
-    content::RenderFrameHost* frame_host,
-    mojo::PendingReceiver<screen_ai::mojom::ScreenAIAnnotator> receiver) {
-  content::BrowserContext* browser_context =
-      frame_host->GetProcess()->GetBrowserContext();
-
-  screen_ai::ScreenAIServiceRouterFactory::GetForBrowserContext(browser_context)
-      ->BindScreenAIAnnotator(std::move(receiver));
-}
-
-void BindScreen2xMainContentExtractor(
-    content::RenderFrameHost* frame_host,
-    mojo::PendingReceiver<screen_ai::mojom::Screen2xMainContentExtractor>
-        receiver) {
-  screen_ai::ScreenAIServiceRouterFactory::GetForBrowserContext(
-      frame_host->GetProcess()->GetBrowserContext())
-      ->BindMainContentExtractor(std::move(receiver));
-}
-#endif
 
 void BindModelBroker(
     content::RenderFrameHost* frame_host,
@@ -579,10 +556,6 @@ void PopulateChromeFrameBinders(
         &web_app::SubAppsServiceImpl::CreateIfAllowed);
   }
 
-  map->Add<screen_ai::mojom::ScreenAIAnnotator>(&BindScreenAIAnnotator);
-
-  map->Add<screen_ai::mojom::Screen2xMainContentExtractor>(
-      &BindScreen2xMainContentExtractor);
 #endif
 
 #if BUILDFLAG(IS_WIN)
diff -up chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders_webui_parts_desktop.cc.disable-ai chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders_webui_parts_desktop.cc
--- chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders_webui_parts_desktop.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/chrome_browser_interface_binders_webui_parts_desktop.cc	2026-10-20 17:09:47.121399478 +0200
@@ -456,9 +456,6 @@ void PopulateChromeWebUIFrameBindersPart
       ::mojom::app_service_internals::AppServiceInternalsPageHandler,
       AppServiceInternalsUI>(map);
 
-  RegisterWebUIControllerInterfaceBinder<
-      ::autofill_ml_internals::mojom::PageHandlerFactory,
-      AutofillMlInternalsUI>(map);
 
   RegisterWebUIControllerInterfaceBinder<
       access_code_cast::mojom::PageHandlerFactory,
diff -up chromium-156.0.8078.35/chrome/browser/chrome_content_browser_client.cc.disable-ai chromium-156.0.8078.35/chrome/browser/chrome_content_browser_client.cc
--- chromium-156.0.8078.35/chrome/browser/chrome_content_browser_client.cc.disable-ai	2026-10-20 17:09:46.915549911 +0200
+++ chromium-156.0.8078.35/chrome/browser/chrome_content_browser_client.cc	2026-10-20 17:09:47.122250321 +0200
@@ -5686,9 +5686,6 @@ ChromeContentBrowserClient::CreateCommit
       &conditions);
 #endif
 
-  MaybeAddCondition(actor::ActorCommitDeferringCondition::MaybeCreate(
-                        *navigation_handle, navigation_type),
-                    &conditions);
 
   return conditions;
 }
diff -up chromium-156.0.8078.35/chrome/browser/component_updater/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/component_updater/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/component_updater/BUILD.gn.disable-ai	2026-10-20 17:09:46.918035707 +0200
+++ chromium-156.0.8078.35/chrome/browser/component_updater/BUILD.gn	2026-10-20 17:09:47.123077844 +0200
@@ -225,13 +225,6 @@ source_set("impl") {
     ]
   }
 
-  if (!is_android && !is_chromeos) {
-    sources += [ "screen_ai_component_installer.cc" ]
-    deps += [
-      "//chrome/browser/screen_ai:screen_ai_install_state",
-      "//services/screen_ai/public/cpp:utilities",
-    ]
-  }
 
   if (use_minikin_hyphenation) {
     sources += [ "hyphenation_component_installer.cc" ]
diff -up chromium-156.0.8078.35/chrome/browser/component_updater/registration.cc.disable-ai chromium-156.0.8078.35/chrome/browser/component_updater/registration.cc
--- chromium-156.0.8078.35/chrome/browser/component_updater/registration.cc.disable-ai	2026-10-20 17:09:46.794353663 +0200
+++ chromium-156.0.8078.35/chrome/browser/component_updater/registration.cc	2026-10-20 17:09:47.123234894 +0200
@@ -64,8 +64,6 @@
 
 #if BUILDFLAG(IS_ANDROID)
 #include "chrome/browser/component_updater/real_time_url_checks_allowlist_component_installer.h"
-#else
-#include "chrome/browser/component_updater/screen_ai_component_installer.h"
 #endif  // BUILDFLAG(IS_ANDROID)
 
 #if !BUILDFLAG(IS_ANDROID)
@@ -202,9 +200,6 @@ void RegisterComponentsForUpdate() {
   RegisterRealTimeUrlChecksAllowlistComponent(cus);
 #endif  // BUIDLFLAG(IS_ANDROID)
 
-#if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
-  ManageScreenAIComponentRegistration(cus, g_browser_process->local_state());
-#endif  // !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
 
   RegisterCommerceHeuristicsComponent(cus);
 
diff -up chromium-156.0.8078.35/chrome/browser/glic/glic_selection_observer.cc.disable-ai chromium-156.0.8078.35/chrome/browser/glic/glic_selection_observer.cc
--- chromium-156.0.8078.35/chrome/browser/glic/glic_selection_observer.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/glic/glic_selection_observer.cc	2026-10-20 17:09:47.123658022 +0200
@@ -1269,20 +1269,6 @@ void GlicSelectionObserver::OnAskGeminiW
   }
 
   std::string skill_prompt;
-  auto* tab_interface =
-      tabs::TabInterface::MaybeGetFromContents(web_contents());
-  if (tab_interface) {
-    if (auto* observer = skills::SkillsUpdateObserver::From(tab_interface)) {
-      if (const auto* list = observer->contextual_skills()) {
-        for (const auto& s : list->skills()) {
-          if (s.id() == skill.id) {
-            skill_prompt = s.prompt();
-            break;
-          }
-        }
-      }
-    }
-  }
 
   if (skill_prompt.empty()) {
     Profile* profile =
@@ -1303,28 +1289,6 @@ void GlicSelectionObserver::OnAskGeminiW
 std::vector<GlicSelectionWidgetDelegate::SkillOption>
 GlicSelectionObserver::GetContextualSkills() {
   std::vector<GlicSelectionWidgetDelegate::SkillOption> result;
-  if (!features::kGlicSelectionPromptSkills.Get()) {
-    return result;
-  }
-  auto* tab_interface =
-      tabs::TabInterface::MaybeGetFromContents(web_contents());
-  if (!tab_interface) {
-    return result;
-  }
-  auto* observer = skills::SkillsUpdateObserver::From(tab_interface);
-  if (!observer) {
-    return result;
-  }
-  const auto* list = observer->contextual_skills();
-  if (!list) {
-    return result;
-  }
-  for (const auto& skill : list->skills()) {
-    if (!skill.id().empty() && !skill.name().empty()) {
-      result.emplace_back(
-          skills::Skill(skill.id(), skill.name(), skill.icon(), ""));
-    }
-  }
   return result;
 }
 
diff -up chromium-156.0.8078.35/chrome/browser/glic/host/glic_skills_manager_impl.cc.disable-ai chromium-156.0.8078.35/chrome/browser/glic/host/glic_skills_manager_impl.cc
--- chromium-156.0.8078.35/chrome/browser/glic/host/glic_skills_manager_impl.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/glic/host/glic_skills_manager_impl.cc	2026-10-20 17:09:47.124113166 +0200
@@ -504,39 +504,6 @@ bool GlicSkillsClientSession::Require1PS
 
 void GlicSkillsClientSession::UpdateSkillPreviews(
     std::optional<tabs::TabInterface*> updated_tab) {
-  if (!skills::SkillsServiceFactory::IsSkillsEnabledForProfile(
-          manager_->profile())) {
-    return;
-  }
-  if (!client_) {
-    return;
-  }
-  auto* active_tab = manager_->active_tab_tracker().GetActiveTab();
-  if (!active_tab ||
-      !manager_->instance().GetSharingManager()->IsTabShared(active_tab)) {
-    client_->NotifyContextualSkillPreviewsChanged({});
-    contextual_skill_previews_.clear();
-    return;
-  }
-  if (updated_tab && active_tab != *updated_tab) {
-    return;
-  }
-  auto* observer = skills::SkillsUpdateObserver::From(active_tab);
-  if (!observer) {
-    return;
-  }
-  auto new_skill_previews = observer->GetContextualSkillPreviews();
-
-  if (mojo::Equals(contextual_skill_previews_, new_skill_previews)) {
-    return;
-  }
-  contextual_skill_previews_ = std::move(new_skill_previews);
-
-  std::vector<mojom::SkillPreviewPtr> skill_previews;
-  for (const auto& preview : contextual_skill_previews_) {
-    skill_previews.push_back(preview.Clone());
-  }
-  client_->NotifyContextualSkillPreviewsChanged(std::move(skill_previews));
 }
 
 void GlicSkillsClientSession::NotifyContextualSkillsChanged(
diff -up chromium-156.0.8078.35/chrome/browser/navigation_predictor/navigation_predictor.cc.disable-ai chromium-156.0.8078.35/chrome/browser/navigation_predictor/navigation_predictor.cc
--- chromium-156.0.8078.35/chrome/browser/navigation_predictor/navigation_predictor.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/navigation_predictor/navigation_predictor.cc	2026-10-20 17:09:47.124547431 +0200
@@ -18,8 +18,6 @@
 #include "base/time/default_tick_clock.h"
 #include "chrome/browser/navigation_predictor/navigation_predictor_keyed_service.h"
 #include "chrome/browser/navigation_predictor/navigation_predictor_keyed_service_factory.h"
-#include "chrome/browser/navigation_predictor/preloading_model_keyed_service.h"
-#include "chrome/browser/navigation_predictor/preloading_model_keyed_service_factory.h"
 #include "chrome/browser/preloading/preloading_prefs.h"
 #include "chrome/browser/profiles/profile.h"
 #include "components/no_state_prefetch/browser/no_state_prefetch_manager.h"
@@ -105,10 +103,6 @@ base::TimeDelta MLModelExecutionTimerSta
       blink::features::kPreloadingModelTimerStartDelay.Get());
 }
 
-base::TimeDelta MLModelExecutionTimerInterval() {
-  return base::Milliseconds(
-      blink::features::kPreloadingModelTimerInterval.Get());
-}
 
 base::TimeDelta MLModelMaxHoverTime() {
   return blink::features::kPreloadingModelMaxHoverTime.Get();
@@ -415,80 +409,6 @@ void NavigationPredictor::ProcessPointer
 }
 
 void NavigationPredictor::OnMLModelExecutionTimerFired() {
-  // Check whether preloading is enabled or not.
-  Profile* profile =
-      Profile::FromBrowserContext(render_frame_host().GetBrowserContext());
-  if (prefetch::IsSomePreloadingEnabled(*profile->GetPrefs()) !=
-      content::PreloadingEligibility::kEligible) {
-    return;
-  }
-
-  // Execute the model.
-  PreloadingModelKeyedService* model_service =
-      PreloadingModelKeyedServiceFactory::GetForProfile(profile);
-  if (!model_service) {
-    return;
-  }
-
-  if (!ml_model_candidate_.has_value()) {
-    return;
-  }
-  auto it = anchors_.find(ml_model_candidate_.value());
-  if (it == anchors_.end()) {
-    return;
-  }
-
-  AnchorElementData& anchor = it->second;
-
-  PreloadingModelKeyedService::Inputs inputs;
-  inputs.contains_image = anchor.contains_image;
-  inputs.font_size = anchor.font_size;
-  inputs.has_text_sibling = anchor.has_text_sibling;
-  inputs.is_bold = anchor.is_bold_font;
-  inputs.is_in_iframe = anchor.is_in_iframe;
-  inputs.is_url_incremented_by_one = anchor.is_url_incremented_by_one;
-  inputs.navigation_start_to_link_logged =
-      anchor.first_report_timestamp - navigation_start_;
-  auto path_info = GetUrlPathLengthDepthAndHash(anchor.target_url);
-  inputs.path_length = path_info.path_length;
-  inputs.path_depth = path_info.path_depth;
-  inputs.percent_clickable_area = anchor.ratio_area;
-  inputs.percent_vertical_distance =
-      static_cast<int>(anchor.ratio_distance_root_top * 100);
-
-  inputs.is_same_host = anchor.is_same_host;
-  auto to_timedelta = [this](std::optional<base::TimeTicks> ts) {
-    return ts.has_value() ? NowTicks() - ts.value() : base::TimeDelta();
-  };
-  // TODO(329691634): Using the real viewport entry time for
-  // `entered_viewport_to_left_viewport` produces low quality results.
-  // We could remove it from the model, if we can't get this to be useful.
-  inputs.entered_viewport_to_left_viewport = base::TimeDelta();
-  inputs.hover_dwell_time = to_timedelta(anchor.pointer_over_timestamp);
-  inputs.pointer_hovering_over_count = anchor.pointer_hovering_over_count;
-  if (model_score_callback_) {
-    std::move(model_score_callback_).Run(inputs);
-  }
-
-  content::PreloadingData* preloading_data =
-      content::PreloadingData::GetOrCreateForWebContents(
-          content::WebContents::FromRenderFrameHost(&render_frame_host()));
-  preloading_data->OnPreloadingHeuristicsModelInput(
-      anchor.target_url,
-      base::BindOnce(&RecordMetricsForModelTraining, inputs,
-                     render_frame_host().GetPageUkmSourceId()));
-  model_service->Score(
-      &scoring_model_task_tracker_, inputs,
-      base::BindOnce(&NavigationPredictor::OnPreloadingHeuristicsModelDone,
-                     weak_ptr_factory_.GetWeakPtr(), anchor.target_url));
-
-  if (inputs.hover_dwell_time < MLModelMaxHoverTime() &&
-      !ml_model_execution_timer_.IsRunning()) {
-    ml_model_execution_timer_.Start(
-        FROM_HERE, MLModelExecutionTimerInterval(),
-        base::BindOnce(&NavigationPredictor::OnMLModelExecutionTimerFired,
-                       base::Unretained(this)));
-  }
 }
 
 void NavigationPredictor::SetModelScoreCallbackForTesting(
diff -up chromium-156.0.8078.35/chrome/browser/page_content_annotations/page_content_annotations_service_factory.cc.disable-ai chromium-156.0.8078.35/chrome/browser/page_content_annotations/page_content_annotations_service_factory.cc
--- chromium-156.0.8078.35/chrome/browser/page_content_annotations/page_content_annotations_service_factory.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/page_content_annotations/page_content_annotations_service_factory.cc	2026-10-20 17:09:47.124907010 +0200
@@ -120,62 +120,6 @@ PageContentAnnotationsServiceFactory::~P
 std::unique_ptr<KeyedService>
 PageContentAnnotationsServiceFactory::BuildServiceInstanceForBrowserContext(
     content::BrowserContext* context) const {
-  Profile* profile = Profile::FromBrowserContext(context);
-
-  if (!ShouldEnablePageContentAnnotations(profile)) {
-    return nullptr;
-  }
-
-  auto* proto_db_provider = profile->GetOriginalProfile()
-                                ->GetDefaultStoragePartition()
-                                ->GetProtoDatabaseProvider();
-  base::FilePath profile_path = profile->GetOriginalProfile()->GetPath();
-
-  // The optimization guide and history services must be available for the page
-  // content annotations service to work.
-  OptimizationGuideKeyedService* optimization_guide_keyed_service =
-      OptimizationGuideKeyedServiceFactory::GetForProfile(profile);
-  history::HistoryService* history_service =
-      HistoryServiceFactory::GetForProfile(profile,
-                                           ServiceAccessType::IMPLICIT_ACCESS);
-  TemplateURLService* template_url_service =
-      TemplateURLServiceFactory::GetForProfile(profile);
-  ZeroSuggestCacheService* zero_suggest_cache_service =
-      ZeroSuggestCacheServiceFactory::GetForProfile(profile);
-
-  if (optimization_guide_keyed_service && history_service) {
-    std::string country_code =
-        GetCurrentCountryCode(g_browser_process->variations_service());
-
-    page_content_annotations::PageEmbeddingsService* page_embeddings_service =
-        page_content_annotations::PageEmbeddingsServiceFactory::GetForProfile(
-            profile);
-    passage_embeddings::EmbedderMetadataProvider* embedder_metadata_provider =
-        page_embeddings_service
-            ? page_embeddings_service->GetEmbedderMetadataProvider()
-            : nullptr;
-
-    auto service = std::make_unique<
-        page_content_annotations::PageContentAnnotationsService>(
-        g_browser_process->GetApplicationLocale(), country_code,
-        optimization_guide_keyed_service, history_service, template_url_service,
-        zero_suggest_cache_service, proto_db_provider, profile_path,
-        optimization_guide_keyed_service->GetOptimizationGuideLogger(),
-        optimization_guide_keyed_service, embedder_metadata_provider,
-        base::ThreadPool::CreateSequencedTaskRunner(
-            {base::MayBlock(), base::TaskPriority::BEST_EFFORT}));
-
-    page_content_annotations::OnDeviceCategoryClassifier* category_classifier =
-        service->on_device_category_classifier();
-    if (category_classifier && page_embeddings_service) {
-      service->SetPageCategoryClassifierBridge(
-          std::make_unique<
-              page_content_annotations::PageCategoryClassifierBridgeImpl>(
-              *page_embeddings_service, *category_classifier));
-    }
-
-    return service;
-  }
   return nullptr;
 }
 
diff -up chromium-156.0.8078.35/chrome/browser/password_manager/password_change_delegate_impl.cc.disable-ai chromium-156.0.8078.35/chrome/browser/password_manager/password_change_delegate_impl.cc
--- chromium-156.0.8078.35/chrome/browser/password_manager/password_change_delegate_impl.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/password_manager/password_change_delegate_impl.cc	2026-10-20 17:09:47.125266785 +0200
@@ -342,12 +342,6 @@ void PasswordChangeDelegateImpl::StartPa
           &PasswordChangeDelegateImpl::OnLoginStateCheckedWithoutPIResult,
           weak_ptr_factory_.GetWeakPtr()));
 
-  // This creates FieldClassificationModelHandler and should trigger
-  // download of a local ML model for field classification.
-  // TODO(452883239): Clean this up when model is downloaded on start-up for
-  // everybody.
-  PasswordFieldClassificationModelHandlerFactory::GetForBrowserContext(
-      originator_->GetBrowserContext());
 }
 
 void PasswordChangeDelegateImpl::OnLoginStateCheckedWithoutPIResult(
diff -up chromium-156.0.8078.35/chrome/browser/pdf/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/pdf/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/pdf/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/pdf/BUILD.gn	2026-10-20 17:09:47.125608663 +0200
@@ -251,10 +251,7 @@ if (!is_android) {
       deps += [
         "//chrome/browser/download:test_support",
         "//chrome/browser/permissions",
-        "//chrome/browser/screen_ai:screen_ai_install_state",
-        "//chrome/browser/screen_ai:screen_ai_service_router_factory",
         "//chrome/browser/ui:browser_manager_service",
-        "//services/screen_ai/public/cpp:utilities",
         "//ui/accessibility:test_support",
       ]
 
diff -up chromium-156.0.8078.35/chrome/browser/pdf/pdf_extension_util.cc.disable-ai chromium-156.0.8078.35/chrome/browser/pdf/pdf_extension_util.cc
--- chromium-156.0.8078.35/chrome/browser/pdf/pdf_extension_util.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/pdf/pdf_extension_util.cc	2026-10-20 17:09:47.125931368 +0200
@@ -142,7 +142,6 @@ base::DictValue GetPdfViewerStrings() {
       {"rotationStateLabel90", IDS_PDF_ROTATION_STATE_LABEL_90},
       {"rotationStateLabel180", IDS_PDF_ROTATION_STATE_LABEL_180},
       {"rotationStateLabel270", IDS_PDF_ROTATION_STATE_LABEL_270},
-      {"searchifyInProgress", IDS_PDF_SEARCHIFY_IN_PROGRESS},
       {"sidebarLabel", IDS_PDF_SIDEBAR_LABEL},
       {"thumbnailPageAriaLabel", IDS_PDF_THUMBNAIL_PAGE_ARIA_LABEL},
       {"tooltipAttachments", IDS_PDF_TOOLTIP_ATTACHMENTS},
diff -up chromium-156.0.8078.35/chrome/browser/permissions/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/permissions/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/permissions/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/permissions/BUILD.gn	2026-10-20 17:09:47.126341654 +0200
@@ -45,12 +45,6 @@ source_set("permissions") {
     "prediction_service/language_detection_observer.h",
     "prediction_service/passage_embedder_delegate.cc",
     "prediction_service/passage_embedder_delegate.h",
-    "prediction_service/permissions_ai_ui_selector.cc",
-    "prediction_service/permissions_ai_ui_selector.h",
-    "prediction_service/prediction_model_handler_provider.cc",
-    "prediction_service/prediction_model_handler_provider.h",
-    "prediction_service/prediction_model_handler_provider_factory.cc",
-    "prediction_service/prediction_model_handler_provider_factory.h",
     "prediction_service/prediction_service_factory.cc",
     "prediction_service/prediction_service_factory.h",
     "prediction_service/prediction_service_request.cc",
diff -up chromium-156.0.8078.35/chrome/browser/permissions/chrome_permissions_client.cc.disable-ai chromium-156.0.8078.35/chrome/browser/permissions/chrome_permissions_client.cc
--- chromium-156.0.8078.35/chrome/browser/permissions/chrome_permissions_client.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/permissions/chrome_permissions_client.cc	2026-10-20 17:09:47.126790464 +0200
@@ -32,7 +32,6 @@
 #include "chrome/browser/permissions/permission_actions_history_factory.h"
 #include "chrome/browser/permissions/permission_decision_auto_blocker_factory.h"
 #include "chrome/browser/permissions/permission_revocation_request.h"
-#include "chrome/browser/permissions/prediction_service/permissions_ai_ui_selector.h"
 #include "chrome/browser/permissions/pref_based_quiet_permission_ui_selector.h"
 #include "chrome/browser/permissions/quiet_notification_permission_ui_config.h"
 #include "chrome/browser/permissions/system/system_permission_settings.h"
@@ -554,8 +553,6 @@ ChromePermissionsClient::CreatePermissio
 #endif
   selectors.emplace_back(std::make_unique<PrefBasedQuietPermissionUiSelector>(
       Profile::FromBrowserContext(browser_context)));
-  selectors.emplace_back(std::make_unique<PermissionsAiUiSelector>(
-      Profile::FromBrowserContext(browser_context)));
   return selectors;
 }
 
@@ -577,8 +574,6 @@ void ChromePermissionsClient::OnPromptRe
 
   Profile* profile =
       Profile::FromBrowserContext(web_contents->GetBrowserContext());
-  PermissionActionsHistoryFactory::GetForProfile(profile)->RecordAction(
-      action, request_type, prompt_disposition);
 
   if (request_type == permissions::RequestType::kNotifications) {
     if (action == permissions::PermissionAction::GRANTED &&
diff -up chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/passage_embedder_delegate.cc.disable-ai chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/passage_embedder_delegate.cc
--- chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/passage_embedder_delegate.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/passage_embedder_delegate.cc	2026-10-20 17:09:47.127337631 +0200
@@ -26,18 +26,6 @@ PassageEmbedderDelegate::PassageEmbedder
     : profile_(profile) {}
 
 Embedder* PassageEmbedderDelegate::GetPassageEmbedder() {
-  if (auto* prediction_model_handler_provider =
-          PredictionModelHandlerProviderFactory::GetForBrowserContext(
-              profile_)) {
-    bool is_ready = prediction_model_handler_provider->IsPassageEmbedderReady();
-    PermissionUmaUtil::RecordPassageEmbedderMetadataValid(is_ready);
-    if (!is_ready) {
-      VLOG(1) << "[PermissionsAIv4] "
-                 "PassageEmbedderDelegate::GetPassageEmbedder is not ready.";
-      return nullptr;
-    }
-    return prediction_model_handler_provider->GetPassageEmbedder();
-  }
   PermissionUmaUtil::RecordPassageEmbedderMetadataValid(false);
   return nullptr;
 }
diff -up chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/permissions_ai_ui_selector.cc.disable-ai chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/permissions_ai_ui_selector.cc
--- chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/permissions_ai_ui_selector.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/permissions/prediction_service/permissions_ai_ui_selector.cc	2026-10-20 17:09:47.127753162 +0200
@@ -810,7 +810,7 @@ void PermissionsAiUiSelector::ExecuteOnD
     ModelExecutionData model_data) {
   VLOG(1) << "[PermissionsAI] ExecuteOnDeviceAivXModel";
   PredictionModelHandlerProvider* prediction_model_handler_provider =
-      PredictionModelHandlerProviderFactory::GetForBrowserContext(profile_);
+      nullptr;
   if (prediction_model_handler_provider) {
     permissions::RequestType request_type =
         model_data.request_metadata.request_type;
diff -up chromium-156.0.8078.35/chrome/browser/prefs/browser_prefs.cc.disable-ai chromium-156.0.8078.35/chrome/browser/prefs/browser_prefs.cc
--- chromium-156.0.8078.35/chrome/browser/prefs/browser_prefs.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/prefs/browser_prefs.cc	2026-10-20 17:09:47.129034282 +0200
@@ -1788,9 +1788,6 @@ void RegisterLocalState(PrefRegistrySimp
   DeviceOAuth2TokenStoreDesktop::RegisterPrefs(registry);
 #endif
 
-#if !BUILDFLAG(IS_ANDROID)
-  screen_ai::RegisterLocalStatePrefs(registry);
-#endif  // !BUILDFLAG(IS_ANDROID)
 
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_ANDROID)
   PlatformAuthPolicyObserver::RegisterPrefs(registry);
diff -up chromium-156.0.8078.35/chrome/browser/prefs/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/prefs/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/prefs/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/prefs/BUILD.gn	2026-10-20 17:09:47.128291911 +0200
@@ -389,7 +389,6 @@ source_set("impl") {
       "//chrome/browser/desktop_to_mobile_promos:utils",
       "//chrome/browser/device_identity",
       "//chrome/browser/new_tab_page",
-      "//chrome/browser/screen_ai:prefs",
       "//chrome/browser/search_engine_choice",
       "//chrome/browser/search_engines",
       "//chrome/browser/sharing_hub",
diff -up chromium-156.0.8078.35/chrome/browser/profiles/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/profiles/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/profiles/BUILD.gn.disable-ai	2026-10-20 17:09:46.920803931 +0200
+++ chromium-156.0.8078.35/chrome/browser/profiles/BUILD.gn	2026-10-20 17:09:47.129488889 +0200
@@ -915,7 +915,6 @@ source_set("profiles_extra_parts_impl")
   } else {
     deps += [
       "//chrome/browser/accessibility/phrase_segmentation",
-      "//chrome/browser/accessibility/tree_fixing:service",
       "//chrome/browser/apps/app_service",
       "//chrome/browser/autofill",
       "//chrome/browser/badging",
@@ -930,7 +929,6 @@ source_set("profiles_extra_parts_impl")
       "//chrome/browser/profile_resetter",
       "//chrome/browser/record_replay",
       "//chrome/browser/record_replay:impl",
-      "//chrome/browser/screen_ai:screen_ai_service_router_factory",
       "//chrome/browser/search",
       "//chrome/browser/search_engine_choice",
       "//chrome/browser/sharing_hub",
diff -up chromium-156.0.8078.35/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc.disable-ai chromium-156.0.8078.35/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
--- chromium-156.0.8078.35/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc.disable-ai	2026-10-20 17:09:46.921544684 +0200
+++ chromium-156.0.8078.35/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc	2026-10-20 17:09:47.132022027 +0200
@@ -134,7 +134,6 @@
 #include "chrome/browser/metrics/profile_metrics_service_factory.h"
 #include "chrome/browser/metrics/variations/google_groups_manager_factory.h"
 #include "chrome/browser/navigation_predictor/navigation_predictor_keyed_service_factory.h"
-#include "chrome/browser/navigation_predictor/preloading_model_keyed_service_factory.h"
 #include "chrome/browser/navigation_predictor/search_engine_preconnector.h"
 #include "chrome/browser/navigation_predictor/search_engine_preconnector_keyed_service_factory.h"
 #include "chrome/browser/net/dns_probe_service_factory.h"
@@ -395,7 +394,6 @@
 #include "chrome/browser/record_replay/task_parameters_extractor_factory.h"
 #include "chrome/browser/record_replay/task_service_factory.h"
 #include "chrome/browser/record_replay/task_store_factory.h"
-#include "chrome/browser/screen_ai/screen_ai_service_router_factory.h"
 #include "chrome/browser/search/instant_service_factory.h"
 #include "chrome/browser/search_engine_choice/search_engine_choice_dialog_service_factory.h"
 #include "chrome/browser/speech/speech_recognition_client_browser_interface_factory.h"
@@ -1185,7 +1183,6 @@ void ChromeBrowserMainExtraPartsProfiles
 #if BUILDFLAG(CHROME_ROOT_STORE_CERT_MANAGEMENT_UI)
   net::ServerCertificateDatabaseServiceFactory::GetInstance();
 #endif
-  PreloadingModelKeyedServiceFactory::GetInstance();
 #if BUILDFLAG(ENABLE_DICE_SUPPORT)
   HistorySyncOptinServiceFactory::GetInstance();
   ProfileManagementDisclaimerServiceFactory::GetInstance();
@@ -1248,9 +1245,6 @@ void ChromeBrowserMainExtraPartsProfiles
   // TODO(b/502297163): Implement for Android.
   OneTimePermissionsTrackerFactory::GetInstance();
 #endif
-  if (optimization_guide::ShouldStartModelValidator()) {
-    optimization_guide::ModelValidatorKeyedServiceFactory::GetInstance();
-  }
   OptimizationGuideGlobalStateHolderKeyedServiceFactory::GetInstance();
   OptimizationGuideKeyedServiceFactory::GetInstance();
   origin_gating::OriginGatingServiceFactory::GetInstance();
@@ -1348,7 +1342,6 @@ void ChromeBrowserMainExtraPartsProfiles
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
   policy::UserPolicyOidcSigninServiceFactory::GetInstance();
 #endif
-  PredictionModelHandlerProviderFactory::GetInstance();
   PredictionServiceFactory::GetInstance();
   predictors::AutocompleteActionPredictorFactory::GetInstance();
   predictors::LoadingPredictorFactory::GetInstance();
@@ -1431,10 +1424,6 @@ void ChromeBrowserMainExtraPartsProfiles
   SafetyHubMenuNotificationServiceFactory::GetInstance();
 #if !BUILDFLAG(IS_ANDROID)
   SafetyHubHatsServiceFactory::GetInstance();
-  if (features::IsMainNodeAnnotationsEnabled()) {
-    screen_ai::AXMainNodeAnnotatorControllerFactory::GetInstance();
-  }
-  screen_ai::ScreenAIServiceRouterFactory::EnsureFactoryBuilt();
 #endif
 #if BUILDFLAG(IS_CHROMEOS)
   if (ash::features::IsScannerEnabled()) {
@@ -1558,7 +1547,6 @@ void ChromeBrowserMainExtraPartsProfiles
   TopSitesFactory::GetInstance();
   translate::TranslateRankerFactory::GetInstance();
 #if !BUILDFLAG(IS_ANDROID)
-  tree_fixing::AXTreeFixingServicesRouterFactory::GetInstance();
   TriggeredProfileResetterFactory::GetInstance();
 #endif
   ttc::TtcKeyedServiceFactory::GetInstance();
diff -up chromium-156.0.8078.35/chrome/browser/profiles/off_the_record_profile_impl.cc.disable-ai chromium-156.0.8078.35/chrome/browser/profiles/off_the_record_profile_impl.cc
--- chromium-156.0.8078.35/chrome/browser/profiles/off_the_record_profile_impl.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/profiles/off_the_record_profile_impl.cc	2026-10-20 17:09:47.132825618 +0200
@@ -229,11 +229,6 @@ void OffTheRecordProfileImpl::Init() {
   // AccessibilityLabelsService has a default prefs behavior in incognito.
   AccessibilityLabelsService::InitOffTheRecordPrefs(this);
 
-#if !BUILDFLAG(IS_ANDROID)
-  // To avoid using any server-side tree fixing service, it is disabled in
-  // Incognito profiles.
-  tree_fixing::InitOffTheRecordPrefs(this);
-#endif  // !BUILDFLAG(IS_ANDROID)
 
   // The ad service might not be available for some irregular profiles, like the
   // System Profile.
diff -up chromium-156.0.8078.35/chrome/browser/profiles/profile_impl.cc.disable-ai chromium-156.0.8078.35/chrome/browser/profiles/profile_impl.cc
--- chromium-156.0.8078.35/chrome/browser/profiles/profile_impl.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/profiles/profile_impl.cc	2026-10-20 17:09:47.133708564 +0200
@@ -923,11 +923,6 @@ void ProfileImpl::DoFinalInit(CreateMode
   // The password settings service needs to start listening to settings
   // changes from Google Mobile Services, as early as possible.
   PasswordManagerSettingsServiceFactory::GetForProfile(this);
-#else
-
-  if (features::IsMainNodeAnnotationsEnabled()) {
-    screen_ai::AXMainNodeAnnotatorControllerFactory::GetForProfile(this);
-  }
 #endif  // BUILDFLAG(IS_ANDROID)
 
   // The announcement notification  service might not be available for some
diff -up chromium-156.0.8078.35/chrome/browser/profiles/profile_manager.cc.disable-ai chromium-156.0.8078.35/chrome/browser/profiles/profile_manager.cc
--- chromium-156.0.8078.35/chrome/browser/profiles/profile_manager.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/profiles/profile_manager.cc	2026-10-20 17:09:47.134733395 +0200
@@ -51,7 +51,6 @@
 #include "chrome/browser/buildflags.h"
 #include "chrome/browser/lifetime/application_lifetime.h"
 #include "chrome/browser/navigation_predictor/navigation_predictor_keyed_service_factory.h"
-#include "chrome/browser/navigation_predictor/preloading_model_keyed_service_factory.h"
 #include "chrome/browser/prefs/incognito_mode_prefs.h"
 #include "chrome/browser/profiles/delete_profile_helper.h"
 #include "chrome/browser/profiles/keep_alive/profile_keep_alive_types.h"
@@ -1603,9 +1602,6 @@ void ProfileManager::DoFinalInitForServi
   // Ensure NavigationPredictorKeyedService is started.
   NavigationPredictorKeyedServiceFactory::GetForProfile(profile);
 
-  // Ensure PreloadingModelKeyedService is started.
-  PreloadingModelKeyedServiceFactory::GetForProfile(profile);
-
   IdentityManagerFactory::GetForProfile(profile)->OnNetworkInitialized();
   AccountReconcilorFactory::GetForProfile(profile);
 #if BUILDFLAG(IS_ANDROID)
diff -up chromium-156.0.8078.35/chrome/browser/resources/pdf/pdf_viewer.html.disable-ai chromium-156.0.8078.35/chrome/browser/resources/pdf/pdf_viewer.html
--- chromium-156.0.8078.35/chrome/browser/resources/pdf/pdf_viewer.html.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/resources/pdf/pdf_viewer.html	2026-10-20 17:09:47.136247873 +0200
@@ -102,10 +102,6 @@ import {AnnotationMode} from './constant
       </cr-page-selector>
     ` : ''}
   </if>
-  <cr-toast id="searchifyProgress">
-    <div class="spinner"></div>
-    <span>$i18n{searchifyInProgress}</span>
-  </cr-toast>
 </div>
 
 ${this.showErrorDialog ? html`<viewer-error-dialog id="error-dialog">
diff -up chromium-156.0.8078.35/chrome/browser/skills/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/skills/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/skills/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/skills/BUILD.gn	2026-10-20 17:09:47.136620563 +0200
@@ -62,13 +62,9 @@ source_set("public") {
 source_set("ui") {
   sources = [
     "skills_dialog_launcher.cc",
-    "skills_update_observer.cc",
-    "skills_update_observer.h",
   ]
   if (!is_android) {
     sources += [
-      "skills_ui_tab_controller.cc",
-      "skills_ui_tab_controller.h",
       "skills_ui_window_controller.cc",
     ]
   }
diff -up chromium-156.0.8078.35/chrome/browser/speech/chrome_speech_recognition_service.cc.disable-ai chromium-156.0.8078.35/chrome/browser/speech/chrome_speech_recognition_service.cc
--- chromium-156.0.8078.35/chrome/browser/speech/chrome_speech_recognition_service.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/speech/chrome_speech_recognition_service.cc	2026-10-20 17:09:47.137036664 +0200
@@ -59,18 +59,6 @@ ChromeSpeechRecognitionService::ChromeSp
     soda_installer_observer_.Observe(soda_installer);
   }
 
-  if (base::FeatureList::IsEnabled(
-          media::kLiveCaptionSpeechRecognitionSmallExpertModel)) {
-    auto* model_installer =
-        g_browser_process
-            ? g_browser_process
-                  ->speech_recognition_small_expert_model_installer()
-            : nullptr;
-    if (model_installer) {
-      speech_recognition_small_expert_model_installer_observer_.Observe(
-          model_installer);
-    }
-  }
 }
 
 ChromeSpeechRecognitionService::~ChromeSpeechRecognitionService() = default;
@@ -98,42 +86,6 @@ void ChromeSpeechRecognitionService::Bin
     options->language = language_name;
   }
 
-  if (base::FeatureList::IsEnabled(
-          media::kLiveCaptionSpeechRecognitionSmallExpertModel) &&
-      options &&
-      options->recognizer_client_type ==
-          media::mojom::RecognizerClientType::kLiveCaption &&
-      !options->is_server_based) {
-    auto* installer =
-        g_browser_process
-            ? g_browser_process
-                  ->speech_recognition_small_expert_model_installer()
-            : nullptr;
-    if (installer &&
-        installer->IsSpeechRecognitionSmallExpertModelInstalled()) {
-      Profile* profile = Profile::FromBrowserContext(context_);
-      if (profile && !model_broker_client_) {
-        auto* opt_guide_service =
-            OptimizationGuideKeyedServiceFactory::GetForProfile(profile);
-        if (opt_guide_service) {
-          model_broker_client_ = opt_guide_service->CreateModelBrokerClient();
-        }
-      }
-      if (model_broker_client_) {
-        auto feature = optimization_guide::mojom::OnDeviceFeature::
-            kSpeechRecognitionSmallExpertModel;
-        model_broker_client_->RequestAssetsFor(feature);
-        model_broker_client_->GetSubscriber(feature).WaitForClient(
-            base::BindOnce(
-                &ChromeSpeechRecognitionService::OnModelClientAvailable,
-                weak_factory_.GetWeakPtr(), std::move(receiver),
-                std::move(client), std::move(options),
-                mojo::WrapCallbackWithDefaultInvokeIfNotRun(std::move(callback),
-                                                            false)));
-        return;
-      }
-    }
-  }
 
   BindSodaRecognizer(std::move(receiver), std::move(client), std::move(options),
                      std::move(callback));
diff -up chromium-156.0.8078.35/chrome/browser/speech/speech_recognition_client_browser_interface.cc.disable-ai chromium-156.0.8078.35/chrome/browser/speech/speech_recognition_client_browser_interface.cc
--- chromium-156.0.8078.35/chrome/browser/speech/speech_recognition_client_browser_interface.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/speech/speech_recognition_client_browser_interface.cc	2026-10-20 17:09:47.137351352 +0200
@@ -67,15 +67,6 @@ SpeechRecognitionClientBrowserInterface:
   if (speech::SodaInstaller::GetInstance()) {
     soda_installer_observation_.Observe(speech::SodaInstaller::GetInstance());
   }
-#if !BUILDFLAG(IS_CHROMEOS)
-  if (base::FeatureList::IsEnabled(
-          media::kLiveCaptionSpeechRecognitionSmallExpertModel) &&
-      g_browser_process &&
-      g_browser_process->speech_recognition_small_expert_model_installer()) {
-    speech_recognition_small_expert_model_installer_observation_.Observe(
-        g_browser_process->speech_recognition_small_expert_model_installer());
-  }
-#endif
   MaybeTriggerModelInstall();
 }
 
@@ -162,24 +153,6 @@ void SpeechRecognitionClientBrowserInter
 #if !BUILDFLAG(IS_CHROMEOS)
   if (base::FeatureList::IsEnabled(
           media::kLiveCaptionSpeechRecognitionSmallExpertModel)) {
-    auto* installer =
-        g_browser_process
-            ? g_browser_process
-                  ->speech_recognition_small_expert_model_installer()
-            : nullptr;
-    if (installer &&
-        !installer->IsSpeechRecognitionSmallExpertModelInstalled() &&
-        !installer->IsSpeechRecognitionSmallExpertModelDownloading() &&
-        installer->GetSpeechRecognitionSmallExpertModelState() !=
-            SpeechRecognitionSmallExpertModelInstaller::
-                SpeechRecognitionSmallExpertModelState::kError &&
-        installer->GetSpeechRecognitionSmallExpertModelState() !=
-            SpeechRecognitionSmallExpertModelInstaller::
-                SpeechRecognitionSmallExpertModelState::
-                    kErrorCorruptPersistent) {
-      Profile* profile = Profile::FromBrowserContext(context_);
-      installer->InstallSpeechRecognitionSmallExpertModel(profile);
-    }
     return;
   }
 #endif
@@ -207,10 +180,7 @@ void SpeechRecognitionClientBrowserInter
 #if !BUILDFLAG(IS_CHROMEOS)
   if (base::FeatureList::IsEnabled(
           media::kLiveCaptionSpeechRecognitionSmallExpertModel)) {
-    if (!g_browser_process ||
-        !g_browser_process->speech_recognition_small_expert_model_installer()) {
       return;
-    }
   } else if (!speech::SodaInstaller::GetInstance()) {
     return;
   }
@@ -232,15 +202,6 @@ void SpeechRecognitionClientBrowserInter
 #if !BUILDFLAG(IS_CHROMEOS)
   if (base::FeatureList::IsEnabled(
           media::kLiveCaptionSpeechRecognitionSmallExpertModel)) {
-    auto* installer =
-        g_browser_process
-            ? g_browser_process
-                  ->speech_recognition_small_expert_model_installer()
-            : nullptr;
-    if (installer &&
-        installer->IsSpeechRecognitionSmallExpertModelInstalled()) {
-      is_installed = true;
-    }
   } else {
     if (speech::SodaInstaller::GetInstance()) {
       is_installed = speech::SodaInstaller::GetInstance()->IsSodaInstalled(
diff -up chromium-156.0.8078.35/chrome/browser/ui/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/ui/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/ui/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/BUILD.gn	2026-10-20 17:09:47.138284584 +0200
@@ -2438,7 +2438,6 @@ static_library("ui") {
       "//chrome/browser/lens/region_search",
       "//chrome/browser/reading_list",
       "//chrome/browser/safe_browsing",
-      "//chrome/browser/screen_ai:screen_ai_service_router_factory",
       "//chrome/browser/smart_card",
       "//chrome/browser/support_tool",
       "//chrome/browser/tab_group_sync:utils",
@@ -3934,7 +3933,6 @@ static_library("ui") {
       "//chrome/browser/ui/views/zoom",
       "//chrome/browser/ui/webauthn:impl",
       "//chrome/browser/ui/webui/app_service_internals",
-      "//chrome/browser/ui/webui/autofill_ml_internals",
       "//chrome/browser/ui/webui/side_panel/customize_chrome",
       "//chrome/browser/ui/window_metadata",
       "//chrome/browser/ui/window_name_prompt",
diff -up chromium-156.0.8078.35/chrome/browser/ui/pdf/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/ui/pdf/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/ui/pdf/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/pdf/BUILD.gn	2026-10-20 17:09:47.138822519 +0200
@@ -6,7 +6,6 @@ import("//pdf/features.gni")
 import("//services/screen_ai/buildflags/features.gni")
 
 assert(enable_pdf)
-assert(enable_screen_ai_service)
 
 source_set("pdf") {
   sources = [
@@ -17,7 +16,6 @@ source_set("pdf") {
     "//chrome/browser/download",
     "//chrome/browser/glic",
     "//chrome/browser/pdf",
-    "//chrome/browser/screen_ai:screen_ai_install_state",
     "//chrome/browser/ui/tab_contents",
     "//chrome/browser/ui/user_education",
     "//chrome/common",
diff -up chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.cc.disable-ai chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.cc
--- chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.cc	2026-10-20 17:09:47.139120273 +0200
@@ -229,22 +229,3 @@ void ChromePDFDocumentHelperClient::SetP
   }
 }
 
-void ChromePDFDocumentHelperClient::OnSearchifyStarted(
-    content::RenderFrameHost& render_frame_host) {
-  // Show the promo only when ScreenAI component is available and OCR can be
-  // done.
-  if (!screen_ai::ScreenAIInstallState::GetInstance()->IsComponentAvailable()) {
-    return;
-  }
-  content::WebContents* web_contents = GetWebContentsToUse(render_frame_host);
-  if (!MaybeShowFeaturePromo(feature_engagement::kIPHPdfSearchifyFeature,
-                             web_contents)) {
-    return;
-  }
-  auto* const tab = tabs::TabInterface::MaybeGetFromContents(web_contents);
-  if (!tab) {
-    return;
-  }
-  tab_subscriptions_.push_back(tab->RegisterWillDeactivate(
-      base::BindRepeating(&MaybeHideSearchifyFeaturePromo)));
-}
diff -up chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.h.disable-ai chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.h
--- chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.h.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/pdf/chrome_pdf_document_helper_client.h	2026-10-20 17:09:47.139286624 +0200
@@ -32,7 +32,6 @@ class ChromePDFDocumentHelperClient : pu
   void OnSaveURL() override;
   void SetPluginCanSave(content::RenderFrameHost& render_frame_host,
                         bool can_save) override;
-  void OnSearchifyStarted(content::RenderFrameHost& render_frame_host) override;
 
   void OnPdfTextExtracted(content::GlobalRenderFrameHostId render_frame_host_id,
                           const std::u16string& text);
diff -up chromium-156.0.8078.35/chrome/browser/ui/tabs/public/tab_features.h.disable-ai chromium-156.0.8078.35/chrome/browser/ui/tabs/public/tab_features.h
--- chromium-156.0.8078.35/chrome/browser/ui/tabs/public/tab_features.h.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/tabs/public/tab_features.h	2026-10-20 17:09:47.139705738 +0200
@@ -220,12 +220,6 @@ namespace web_app {
 class WindowManagementContentSettingObserver;
 }  // namespace web_app
 
-#if !BUILDFLAG(IS_ANDROID)
-namespace skills {
-class SkillsUpdateObserver;
-}  // namespace skills
-#endif  // !BUILDFLAG(IS_ANDROID)
-
 namespace sync_sessions {
 class SyncSessionsRouterTabHelper;
 }  // namespace sync_sessions
@@ -779,10 +773,6 @@ class TabFeatures {
   std::unique_ptr<contextual_tasks::ContextualTasksTabVisitTracker>
       contextual_tasks_tab_visit_tracker_;
 
-#if !BUILDFLAG(IS_ANDROID)
-  std::unique_ptr<skills::SkillsUpdateObserver> skills_update_observer_;
-#endif  //  !BUILDFLAG(IS_ANDROID)
-
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
   std::unique_ptr<metrics::DesktopSessionDurationObserver>
       desktop_session_duration_observer_;
diff -up chromium-156.0.8078.35/chrome/browser/ui/tabs/tab_features.cc.disable-ai chromium-156.0.8078.35/chrome/browser/ui/tabs/tab_features.cc
--- chromium-156.0.8078.35/chrome/browser/ui/tabs/tab_features.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/tabs/tab_features.cc	2026-10-20 17:09:47.140387632 +0200
@@ -506,11 +506,6 @@ void TabFeatures::Init(TabInterface& tab
           GetUserDataFactory().CreateInstance<actor::ui::ActorUiTabController>(
               tab, tab, actor::ActorKeyedService::Get(profile));
     }
-    if (base::FeatureList::IsEnabled(features::kSkillsEnabled)) {
-      skills_ui_tab_controller_ =
-          GetUserDataFactory().CreateInstance<skills::SkillsUiTabController>(
-              tab, tab);
-    }
   }  // IsInNormalWindow() end.
 
   if (base::FeatureList::IsEnabled(features::kGlicActor)) {
@@ -814,10 +809,6 @@ void TabFeatures::Init(TabInterface& tab
   }
 #endif
 
-  if (base::FeatureList::IsEnabled(features::kSkillsEnabled)) {
-    skills_update_observer_ =
-        std::make_unique<skills::SkillsUpdateObserver>(tab);
-  }
   if (base::FeatureList::IsEnabled(features::kIndigo)) {
     indigo_page_action_controller_ =
         std::make_unique<indigo::IndigoPageActionController>(
diff -up chromium-156.0.8078.35/chrome/browser/ui/views/user_education/browser_user_education_service.cc.disable-ai chromium-156.0.8078.35/chrome/browser/ui/views/user_education/browser_user_education_service.cc
--- chromium-156.0.8078.35/chrome/browser/ui/views/user_education/browser_user_education_service.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/views/user_education/browser_user_education_service.cc	2026-10-20 17:09:47.141124695 +0200
@@ -1082,40 +1082,6 @@ void MaybeRegisterChromeFeaturePromos(
                        "Triggered when the PDF Viewer opens.")));
 #endif  // BUILDFLAG(ENABLE_PDF_INK2)
 
-  // kIPHPdfSearchifyFeature:
-  registry.RegisterFeature(std::move(
-      FeaturePromoSpecification::CreateForToastPromo(
-          feature_engagement::kIPHPdfSearchifyFeature,
-          kBrowserDialogAnchorElementId, IDS_PDF_SEARCHIFY_IPH_BODY,
-          IDS_PDF_SEARCHIFY_IPH_BODY_SCREEN_READER,
-          FeaturePromoSpecification::AcceleratorInfo())
-          .SetBubbleArrow(HelpBubbleArrow::kNone)
-          .SetBubbleTitleText(IDS_PDF_SEARCHIFY_IPH_TITLE)
-          .SetMetadata(132, "rhalavati@chromium.org",
-                       "Triggered once when user opens a PDF which gets OCRed.")
-          .SetAnchorElementFilter(base::BindRepeating(
-              [](const ui::ElementTracker::ElementList& elements)
-                  -> ui::TrackedElement* {
-                if (elements.empty()) {
-                  return nullptr;
-                }
-                // Ensure a searchified PDF is visible before showing the IPH.
-                auto* const browser_view =
-                    views::ElementTrackerViews::GetInstance()
-                        ->GetFirstMatchingViewAs<BrowserView>(
-                            kBrowserViewElementId, elements[0]->context());
-                std::vector<ContentsWebView*> contents_web_views =
-                    browser_view->GetAllVisibleContentsWebViews();
-                for (auto* contents_web_view : contents_web_views) {
-                  auto* pdf_doc_helper =
-                      pdf::PDFDocumentHelper::MaybeGetForWebContents(
-                          *contents_web_view->GetWebContents());
-                  if (pdf_doc_helper && pdf_doc_helper->SearchifyStarted()) {
-                    return elements[0];
-                  }
-                }
-                return nullptr;
-              }))));
 
   // kIPHLensOverlayFeature:
   registry.RegisterFeature(std::move(
diff -up chromium-156.0.8078.35/chrome/browser/ui/webui/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/ui/webui/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/ui/webui/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/webui/BUILD.gn	2026-10-20 17:09:47.141678392 +0200
@@ -114,7 +114,6 @@ source_set("configs") {
       "//chrome/browser/actor/ui:actor_overlay_web_view",
       "//chrome/browser/ui/webui/access_code_cast",
       "//chrome/browser/ui/webui/app_service_internals",
-      "//chrome/browser/ui/webui/autofill_ml_internals",
       "//chrome/browser/ui/webui/color_pipeline_internals",
       "//chrome/browser/ui/webui/critical_actions",
       "//chrome/browser/ui/webui/infobar_internals",
diff -up chromium-156.0.8078.35/chrome/browser/ui/webui/chrome_web_ui_configs.cc.disable-ai chromium-156.0.8078.35/chrome/browser/ui/webui/chrome_web_ui_configs.cc
--- chromium-156.0.8078.35/chrome/browser/ui/webui/chrome_web_ui_configs.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/webui/chrome_web_ui_configs.cc	2026-10-20 17:09:47.142058709 +0200
@@ -354,7 +354,6 @@ void RegisterChromeWebUIConfigs() {
   map.AddWebUIConfig(std::make_unique<media_router::AccessCodeCastUIConfig>());
   map.AddWebUIConfig(std::make_unique<actor::ui::ActorOverlayUIConfig>());
   map.AddWebUIConfig(std::make_unique<AppServiceInternalsUIConfig>());
-  map.AddWebUIConfig(std::make_unique<AutofillMlInternalsUIConfig>());
   map.AddWebUIConfig(std::make_unique<BookmarksSidePanelUIConfig>());
   map.AddWebUIConfig(std::make_unique<BookmarksUIConfig>());
   map.AddWebUIConfig(std::make_unique<ColorPipelineInternalsUIConfig>());
diff -up chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.cc.disable-ai chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.cc
--- chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.cc	2026-10-20 17:09:47.142771034 +0200
@@ -35,11 +35,6 @@ void AccessibilityMainHandler::RegisterM
       base::BindRepeating(
           &AccessibilityMainHandler::HandleCheckAccessibilityImageLabels,
           base::Unretained(this)));
-  web_ui()->RegisterMessageCallback(
-      "getScreenAiInstallState",
-      base::BindRepeating(
-          &AccessibilityMainHandler::HandleGetScreenAIInstallState,
-          base::Unretained(this)));
 }
 
 void AccessibilityMainHandler::OnJavascriptAllowed() {
@@ -50,11 +45,6 @@ void AccessibilityMainHandler::OnJavascr
           base::Unretained(this)));
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
-  if (features::IsMainNodeAnnotationsEnabled()) {
-    CHECK(!component_ready_observer_.IsObserving());
-    component_ready_observer_.Observe(
-        screen_ai::ScreenAIInstallState::GetInstance());
-  }
 }
 
 void AccessibilityMainHandler::OnJavascriptDisallowed() {
@@ -62,39 +52,6 @@ void AccessibilityMainHandler::OnJavascr
   accessibility_subscription_ = {};
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
-  if (features::IsMainNodeAnnotationsEnabled()) {
-    component_ready_observer_.Reset();
-  }
-}
-
-void AccessibilityMainHandler::DownloadProgressChanged(double progress) {
-  CHECK_GE(progress, 0.0);
-  CHECK_LE(progress, 1.0);
-  const int progress_num = progress * 100;
-  FireWebUIListener("screen-ai-downloading-progress-changed",
-                    base::Value(progress_num));
-}
-
-void AccessibilityMainHandler::StateChanged(
-    screen_ai::ScreenAIInstallState::State state) {
-  base::Value state_value = base::Value(static_cast<int>(state));
-  FireWebUIListener("screen-ai-state-changed", state_value);
-}
-
-void AccessibilityMainHandler::OnScreenAIInstallStateDestroying() {
-  component_ready_observer_.Reset();
-}
-
-void AccessibilityMainHandler::HandleGetScreenAIInstallState(
-    const base::ListValue& args) {
-  CHECK_EQ(1U, args.size());
-  const base::Value& callback_id = args[0];
-  AllowJavascript();
-  // Get the current install state and send it back to a UI callback.
-  screen_ai::ScreenAIInstallState::State current_install_state =
-      screen_ai::ScreenAIInstallState::GetInstance()->get_state();
-  ResolveJavascriptCallback(
-      callback_id, base::Value(static_cast<int>(current_install_state)));
 }
 
 void AccessibilityMainHandler::HandleGetScreenReaderState(
diff -up chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.h.disable-ai chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.h
--- chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.h.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/webui/settings/accessibility_main_handler.h	2026-10-20 17:09:47.142925997 +0200
@@ -19,8 +19,7 @@ namespace settings {
 // Settings handler for the main accessibility settings page,
 // chrome://settings/accessibility.
 class AccessibilityMainHandler
-    : public ::settings::SettingsPageUIHandler,
-      public screen_ai::ScreenAIInstallState::Observer {
+    : public ::settings::SettingsPageUIHandler {
  public:
   AccessibilityMainHandler();
   ~AccessibilityMainHandler() override;
@@ -32,10 +31,6 @@ class AccessibilityMainHandler
   void OnJavascriptAllowed() override;
   void OnJavascriptDisallowed() override;
 
-  // screen_ai::ScreenAIInstallState::Observer:
-  void DownloadProgressChanged(double progress) override;
-  void StateChanged(screen_ai::ScreenAIInstallState::State state) override;
-  void OnScreenAIInstallStateDestroying() override;
 
  private:
   void HandleGetScreenReaderState(const base::ListValue& args);
@@ -52,9 +47,6 @@ class AccessibilityMainHandler
   base::CallbackListSubscription accessibility_subscription_;
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
-  base::ScopedObservation<screen_ai::ScreenAIInstallState,
-                          screen_ai::ScreenAIInstallState::Observer>
-      component_ready_observer_{this};
 };
 
 }  // namespace settings
diff -up chromium-156.0.8078.35/chrome/browser/ui/webui/settings/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/ui/webui/settings/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/ui/webui/settings/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/webui/settings/BUILD.gn	2026-10-20 17:09:47.142455715 +0200
@@ -62,7 +62,6 @@ source_set("settings") {
     "//chrome/browser/permissions/system",
     "//chrome/browser/profile_resetter",
     "//chrome/browser/profiles:profile",
-    "//chrome/browser/screen_ai:screen_ai_install_state",
     "//chrome/browser/themes",
     "//chrome/browser/ui/safety_hub",
     "//chrome/browser/ui/search_engines",
@@ -680,7 +679,6 @@ if (!is_android) {
       ]
       deps += [
         "//chrome/browser/profiles",
-        "//chrome/browser/screen_ai:screen_ai_install_state",
         "//ui/accessibility",
       ]
     }
diff -up chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/BUILD.gn
--- chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/BUILD.gn	2026-10-20 17:09:47.143104559 +0200
@@ -36,7 +36,6 @@ source_set("read_anything") {
     "//chrome/browser/profiles:profile",
     "//chrome/browser/resources/side_panel/read_anything:resources",
     "//chrome/browser/resources/side_panel/shared:resources_grit",
-    "//chrome/browser/screen_ai:screen_ai_service_router_factory",
     "//chrome/browser/speech",
     "//chrome/browser/translate",
     "//chrome/browser/ui:accelerator_table",
diff -up chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc.disable-ai chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc
--- chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_page_handler.cc	2026-10-20 17:09:47.143582610 +0200
@@ -19,8 +19,6 @@
 #include "chrome/browser/dom_distiller/dom_distiller_service_factory.h"
 #include "chrome/browser/language/language_model_manager_factory.h"
 #include "chrome/browser/profiles/profile.h"
-#include "chrome/browser/screen_ai/screen_ai_service_router.h"
-#include "chrome/browser/screen_ai/screen_ai_service_router_factory.h"
 #include "chrome/browser/speech/extension_api/tts_engine_extension_api.h"
 #include "chrome/browser/translate/chrome_translate_client.h"
 #include "chrome/browser/ui/read_anything/read_anything_controller.h"
@@ -478,14 +476,6 @@ ReadAnythingUntrustedPageHandler::ReadAn
   prefs_lang = language::ExtractBaseLanguage(prefs_lang);
   SetDefaultLanguageCode(prefs_lang);
 
-  if (use_screen_ai_service_) {
-    screen_ai::ScreenAIServiceRouterFactory::GetForBrowserContext(profile_)
-        ->GetServiceStateAsync(
-            screen_ai::ScreenAIServiceRouter::Service::kMainContentExtraction,
-            base::BindOnce(
-                &ReadAnythingUntrustedPageHandler::OnScreenAIServiceInitialized,
-                weak_factory_.GetWeakPtr()));
-  }
 
   if (features::IsReadAnythingWithReadabilityEnabled() &&
       !features::IsReadAnythingReadAloudPhraseHighlightingEnabled()) {
diff -up chromium-156.0.8078.35/chrome/renderer/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/renderer/BUILD.gn
--- chromium-156.0.8078.35/chrome/renderer/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/renderer/BUILD.gn	2026-10-20 17:09:47.144053919 +0200
@@ -346,8 +346,6 @@ static_library("renderer") {
     deps += [ "//chrome/common/wallet:mojo_bindings" ]
   } else {
     sources += [
-      "accessibility/ax_tree_distiller.cc",
-      "accessibility/ax_tree_distiller.h",
       "accessibility/phrase_segmentation/base_phrase_segmenter.cc",
       "accessibility/phrase_segmentation/base_phrase_segmenter.h",
       "accessibility/phrase_segmentation/dependency_parser_model.cc",
@@ -364,23 +362,6 @@ static_library("renderer") {
       "accessibility/phrase_segmentation/tokenized_sentence.h",
       "accessibility/phrase_segmentation/tokenizer.cc",
       "accessibility/phrase_segmentation/tokenizer.h",
-      "accessibility/read_anything/read_aloud_app_model.cc",
-      "accessibility/read_anything/read_aloud_app_model.h",
-      "accessibility/read_anything/read_aloud_traversal_utils.cc",
-      "accessibility/read_anything/read_aloud_traversal_utils.h",
-      "accessibility/read_anything/read_anything_app_controller.cc",
-      "accessibility/read_anything/read_anything_app_controller.h",
-      "accessibility/read_anything/read_anything_app_model.cc",
-      "accessibility/read_anything/read_anything_app_model.h",
-      "accessibility/read_anything/read_anything_distiller.h",
-      "accessibility/read_anything/read_anything_distiller_factory.cc",
-      "accessibility/read_anything/read_anything_distiller_factory.h",
-      "accessibility/read_anything/read_anything_node_utils.cc",
-      "accessibility/read_anything/read_anything_node_utils.h",
-      "accessibility/read_anything/read_anything_readability_distiller.cc",
-      "accessibility/read_anything/read_anything_readability_distiller.h",
-      "accessibility/read_anything/read_anything_screen2x_distiller.cc",
-      "accessibility/read_anything/read_anything_screen2x_distiller.h",
       "media/chrome_speech_recognition_client.cc",
       "media/chrome_speech_recognition_client.h",
       "password_manager/remote_actor_credential_sharing_extension.cc",
@@ -398,7 +379,6 @@ static_library("renderer") {
       "//components/record_replay/content/renderer",
       "//components/trusted_vault",
       "//services/screen_ai/buildflags",
-      "//services/screen_ai/public/mojom",
       "//services/strings",
       "//third_party/re2",
       "//third_party/tensorflow-text",
diff -up chromium-156.0.8078.35/chrome/renderer/chrome_render_frame_observer.cc.disable-ai chromium-156.0.8078.35/chrome/renderer/chrome_render_frame_observer.cc
--- chromium-156.0.8078.35/chrome/renderer/chrome_render_frame_observer.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/renderer/chrome_render_frame_observer.cc	2026-10-20 17:09:47.144502641 +0200
@@ -88,7 +88,6 @@
 #include "url/url_constants.h"
 
 #if !BUILDFLAG(IS_ANDROID)
-#include "chrome/renderer/accessibility/read_anything/read_anything_app_controller.h"
 #include "chrome/renderer/searchbox/searchbox_extension.h"
 #endif  // !BUILDFLAG(IS_ANDROID)
 
@@ -385,14 +384,6 @@ void ChromeRenderFrameObserver::DidClear
     SearchBoxExtension::Install(render_frame()->GetWebFrame());
   }
 
-  // Install ReadAnythingAppController on render frames with the Read Anything
-  // url, which is chrome-untrusted. ReadAnythingAppController installs v8
-  // bindings in the chrome.readingMode namespace which are consumed by
-  // read_anything/app.ts, the resource of the Read Anything WebUI.
-  if (render_frame()->GetWebFrame()->GetDocument().Url() ==
-      chrome::kChromeUIUntrustedReadAnythingSidePanelURL) {
-    ReadAnythingAppController::Install(render_frame());
-  }
 #endif  // !BUILDFLAG(IS_ANDROID)
 #if BUILDFLAG(ENABLE_GUEST_VIEW) && !BUILDFLAG(ENABLE_EXTENSIONS_CORE)
   guest_view::SlimWebViewBindings::MaybeInstall(*render_frame());
diff -up chromium-156.0.8078.35/chrome/test/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/test/BUILD.gn
--- chromium-156.0.8078.35/chrome/test/BUILD.gn.disable-ai	2026-10-20 17:09:46.922795453 +0200
+++ chromium-156.0.8078.35/chrome/test/BUILD.gn	2026-10-20 17:09:47.145277166 +0200
@@ -1891,9 +1891,6 @@ if (!is_android) {
       "//chrome/browser/safe_browsing:blocking_page_platform_test_support",
       "//chrome/browser/safe_browsing:browser_tests",
       "//chrome/browser/safe_browsing:verdict_cache_manager_factory",
-      "//chrome/browser/screen_ai:browser_tests",
-      "//chrome/browser/screen_ai:screen_ai_install_state",
-      "//chrome/browser/screen_ai:screen_ai_service_router_factory",
       "//chrome/browser/search",
       "//chrome/browser/search:browser_tests",
       "//chrome/browser/search:test_support",
@@ -2537,8 +2534,6 @@ if (!is_android) {
       "//services/preferences/public/cpp",
       "//services/preferences/public/cpp/tracked",
       "//services/preferences/tracked:features",
-      "//services/screen_ai/public/cpp:utilities",
-      "//services/screen_ai/public/mojom",
       "//services/service_manager/public/cpp",
       "//services/strings",
       "//services/test/echo/public/mojom",
@@ -6309,7 +6304,6 @@ test("unit_tests") {
     # TODO(crbug.com/417513088): Maybe merge with the non-android `deps`
     # declaration above?
     deps += [
-      "../browser/screen_ai:screen_ai_install_state",
       "//chrome:packed_resources",
       "//chrome/browser/apps/app_service",
       "//chrome/browser/apps/app_service:app_registry_cache_waiter",
@@ -6348,7 +6342,6 @@ test("unit_tests") {
       "//chrome/browser/profile_resetter:fake_profile_resetter",
       "//chrome/browser/profile_resetter:unit_tests",
       "//chrome/browser/resources/new_tab_page_instant:resources_grit",
-      "//chrome/browser/screen_ai:unit_tests",
       "//chrome/browser/search/background",
       "//chrome/browser/search_engine_choice:unit_tests",
       "//chrome/browser/sensor:unit_tests",
diff -up chromium-156.0.8078.35/chrome/utility/BUILD.gn.disable-ai chromium-156.0.8078.35/chrome/utility/BUILD.gn
--- chromium-156.0.8078.35/chrome/utility/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/utility/BUILD.gn	2026-10-20 17:09:47.146600184 +0200
@@ -153,7 +153,6 @@ static_library("utility") {
       "//components/user_data_importer/content",
       "//components/user_data_importer/utility:bookmarks",
       "//services/proxy_resolver:lib",
-      "//services/screen_ai",
     ]
   }
 
diff -up chromium-156.0.8078.35/chrome/utility/services.cc.disable-ai chromium-156.0.8078.35/chrome/utility/services.cc
--- chromium-156.0.8078.35/chrome/utility/services.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/chrome/utility/services.cc	2026-10-20 17:09:47.146947839 +0200
@@ -70,8 +70,6 @@
 #include "components/mirroring/service/mirroring_service.h"
 #include "services/proxy_resolver/proxy_resolver_factory_impl.h"  // nogncheck
 #include "services/proxy_resolver/public/mojom/proxy_resolver.mojom.h"
-#include "services/screen_ai/public/mojom/screen_ai_factory.mojom.h"  // nogncheck
-#include "services/screen_ai/screen_ai_service_impl.h"  // nogncheck
 #endif  // !BUILDFLAG(IS_ANDROID)
 
 #if BUILDFLAG(ENABLE_MEDIA_REMOTING_REDIRECTION)
@@ -301,10 +299,6 @@ auto RunReadingModeMetricsService(
       std::move(receiver));
 }
 
-auto RunScreenAIServiceFactory(
-    mojo::PendingReceiver<screen_ai::mojom::ScreenAIServiceFactory> receiver) {
-  return std::make_unique<screen_ai::ScreenAIService>(std::move(receiver));
-}
 #endif
 
 #if (BUILDFLAG(SAFE_BROWSING_DOWNLOAD_PROTECTION) && \
@@ -512,7 +506,6 @@ void RegisterMainThreadServices(mojo::Se
   services.Add(RunProfileImporter);
   services.Add(RunMirroringService);
   services.Add(RunReadingModeMetricsService);
-  services.Add(RunScreenAIServiceFactory);
 #endif  // !BUILDFLAG(IS_ANDROID)
 
 #if BUILDFLAG(ENABLE_MEDIA_REMOTING_REDIRECTION)
diff -up chromium-156.0.8078.35/components/autofill/core/browser/foundations/autofill_manager.cc.disable-ai chromium-156.0.8078.35/components/autofill/core/browser/foundations/autofill_manager.cc
--- chromium-156.0.8078.35/components/autofill/core/browser/foundations/autofill_manager.cc.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/components/autofill/core/browser/foundations/autofill_manager.cc	2026-10-20 17:09:47.147598715 +0200
@@ -164,8 +164,6 @@ struct AutofillManager::AsyncContext {
 
   std::vector<FormData> forms;
   std::vector<RegexPredictions> regex_predictions;
-  std::vector<ModelPredictions> autofill_predictions;
-  std::vector<ModelPredictions> password_manager_predictions;
   GeoIpCountryCode country_code;
   LanguageCode current_page_language;
   std::unique_ptr<BufferingLogManager> log_manager;
@@ -888,10 +886,8 @@ void AutofillManager::RunMlModels(
                                    }));
         switch (source) {
           case HeuristicSource::kAutofillMachineLearning:
-            context.autofill_predictions = std::move(model_predictions);
             break;
           case HeuristicSource::kPasswordManagerMachineLearning:
-            context.password_manager_predictions = std::move(model_predictions);
             break;
           case HeuristicSource::kRegexes:
             NOTREACHED();
@@ -1068,12 +1064,6 @@ void AutofillManager::UpdateFormCache(
 
   auto apply_predictions = [](FormStructure& form_structure,
                               const AsyncContext& context, size_t i) {
-    if (!context.autofill_predictions.empty()) {
-      context.autofill_predictions[i].ApplyTo(form_structure.fields());
-    }
-    if (!context.password_manager_predictions.empty()) {
-      context.password_manager_predictions[i].ApplyTo(form_structure.fields());
-    }
     if (!context.regex_predictions.empty()) {
       context.regex_predictions[i].ApplyTo(form_structure.fields());
     }
@@ -1167,26 +1157,6 @@ void AutofillManager::LogCurrentFieldTyp
 
 void AutofillManager::SubscribeToMlModelChanges(
     FieldClassificationModelHandler& handler) {
-  switch (handler.optimization_target()) {
-    case optimization_guide::proto::OptimizationTarget::
-        OPTIMIZATION_TARGET_AUTOFILL_FIELD_CLASSIFICATION:
-      if (!autofill_model_change_subscription_) {
-        autofill_model_change_subscription_ =
-            handler.RegisterModelChangeCallback(base::BindRepeating(
-                &AutofillManager::ReparseKnownForms, base::Unretained(this)));
-      }
-      break;
-    case optimization_guide::proto::OptimizationTarget::
-        OPTIMIZATION_TARGET_PASSWORD_MANAGER_FORM_CLASSIFICATION:
-      if (!password_manager_model_change_subscription_) {
-        password_manager_model_change_subscription_ =
-            handler.RegisterModelChangeCallback(base::BindRepeating(
-                &AutofillManager::ReparseKnownForms, base::Unretained(this)));
-      }
-      break;
-    default:
-      NOTREACHED();
-  }
 }
 
 }  // namespace autofill
diff -up chromium-156.0.8078.35/components/BUILD.gn.disable-ai chromium-156.0.8078.35/components/BUILD.gn
--- chromium-156.0.8078.35/components/BUILD.gn.disable-ai	2026-10-20 17:09:46.924231964 +0200
+++ chromium-156.0.8078.35/components/BUILD.gn	2026-10-20 17:09:47.147217409 +0200
@@ -639,7 +639,6 @@ test("components_unittests") {
       "//components/one_time_tokens/core/browser:unit_tests",
       "//components/page_info:unit_tests",
       "//components/password_manager/core/browser/actor_login:unit_tests",
-      "//components/permissions/prediction_service:unit_tests",
       "//components/privacy_sandbox/privacy_sandbox_attestations:unit_tests",
       "//components/safety_check:unit_tests",
       "//components/security_interstitials/content:unit_tests",
diff -up chromium-156.0.8078.35/components/page_content_annotations/content/BUILD.gn.disable-ai chromium-156.0.8078.35/components/page_content_annotations/content/BUILD.gn
--- chromium-156.0.8078.35/components/page_content_annotations/content/BUILD.gn.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/components/page_content_annotations/content/BUILD.gn	2026-10-20 17:09:47.148056768 +0200
@@ -75,8 +75,6 @@ source_set("page_embeddings") {
   sources = [
     "embeddings_candidate_generator.cc",
     "embeddings_candidate_generator.h",
-    "page_category_classifier_bridge_impl.cc",
-    "page_category_classifier_bridge_impl.h",
     "page_embeddings_service.cc",
     "page_embeddings_service.h",
   ]
diff -up chromium-156.0.8078.35/components/pdf/renderer/pdf_view_web_plugin_client.h.disable-ai chromium-156.0.8078.35/components/pdf/renderer/pdf_view_web_plugin_client.h
--- chromium-156.0.8078.35/components/pdf/renderer/pdf_view_web_plugin_client.h.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/components/pdf/renderer/pdf_view_web_plugin_client.h	2026-10-20 17:09:47.148412437 +0200
@@ -12,7 +12,6 @@
 #include "base/memory/weak_ptr.h"
 #include "mojo/public/cpp/bindings/remote.h"
 #include "pdf/pdf_view_web_plugin.h"
-#include "services/screen_ai/public/mojom/screen_ai_service.mojom.h"
 
 namespace blink {
 class WebLocalFrame;
@@ -110,8 +109,6 @@ class PdfViewWebPluginClient : public ch
 
   raw_ptr<blink::WebPluginContainer> plugin_container_;
 
-  mojo::Remote<screen_ai::mojom::ScreenAIAnnotator> screen_ai_annotator_;
-  base::RepeatingClosure ocr_disconnect_callback_;
 
   base::WeakPtrFactory<PdfViewWebPluginClient> weak_factory_{this};
 };
diff -up chromium-156.0.8078.35/services/screen_ai/buildflags/features.gni.disable-ai chromium-156.0.8078.35/services/screen_ai/buildflags/features.gni
--- chromium-156.0.8078.35/services/screen_ai/buildflags/features.gni.disable-ai	2026-10-14 00:13:09.000000000 +0200
+++ chromium-156.0.8078.35/services/screen_ai/buildflags/features.gni	2026-10-20 17:09:47.148697659 +0200
@@ -6,15 +6,13 @@ import("//build/config/sanitizers/saniti
 
 declare_args() {
   # Screen AI service is only supported on desktop platforms.
-  enable_screen_ai_service = is_linux || is_mac || is_chromeos || is_win
+  enable_screen_ai_service = false
 
   # Screen AI library is not available for browser tests on ChromeOS. Tast tests
   # cover library's OCR functionality on ChromeOS.
   enable_screen_ai_browsertests =
-      (is_linux && target_cpu == "x64") ||
-      (is_mac && (target_cpu == "arm64" || target_cpu == "x64")) ||
-      (is_win && (target_cpu == "x64" || target_cpu == "x86"))
+      false
 
   # Screen AI library is not available for MSAN and UBSAN.
-  use_fake_screen_ai = is_msan || is_ubsan || is_ubsan_vptr || is_ubsan_security
+  use_fake_screen_ai = false
 }
